test/functional/p2p_orphan_handling.py

```diff
@@ -566,6 +569,47 @@ def test_orphan_txid_inv(self):
         assert tx_child["txid"] in node_mempool
         assert_equal(node.getmempoolentry(tx_child["txid"])["wtxid"], tx_child["wtxid"])
 
+    @cleanup
+    def test_max_orphan_amount(self):
+        self.log.info("Check that we never exceed our storage limits for orphans")
+
+        node = self.nodes[0]
+        self.generate(self.wallet, 1)
+        peer_1 = node.add_p2p_connection(P2PInterface())
+
+        self.log.info("Check that orphanage is empty on start of test")
+        assert len(node.getorphantxs()) == 0
+
+        self.log.info("Filling up orphanage with " + str(DEFAULT_MAX_ORPHAN_TRANSACTIONS) + "(DEFAULT_MAX_ORPHAN_TRANSACTIONS) orphans")
+        orphans = []
+        parent_orphans = []
+        for _ in range(DEFAULT_MAX_ORPHAN_TRANSACTIONS):
+            tx_parent_1 = self.wallet.create_self_transfer()
+            tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
+            parent_orphans.append(tx_parent_1["tx"])
+            orphans.append(tx_child_1["tx"])
+            peer_1.send_message(msg_tx(tx_child_1["tx"]))
+
+        peer_1.sync_with_ping()
+        orphanage = node.getorphantxs()
+        assert_equal(len(orphanage), DEFAULT_MAX_ORPHAN_TRANSACTIONS)
+
+        for orphan in orphans:
+            assert tx_in_orphanage(node, orphan)
+
+        self.log.info("Check that we do not add more than the max orphan amount")
+        tx_parent_1 = self.wallet.create_self_transfer()
+        tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
+        peer_1.send_and_ping(msg_tx(tx_child_1["tx"]))
+        parent_orphans.append(tx_parent_1["tx"])
+        orphanage = node.getorphantxs()
+        assert_equal(len(orphanage), DEFAULT_MAX_ORPHAN_TRANSACTIONS)
```

Comment:
```suggestion
        assert_equal(len(node.getorphantxs()), DEFAULT_MAX_ORPHAN_TRANSACTIONS)
```
Storing in another variable is not necessary only for one time usage.

You don't need to invalidate ACKs only for this though.